src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -210,6 +210,19 @@ internal static byte[] GetEncodedTruncatedBytesFromString(string? text, Encoding
                 return encoding.GetBytes(text);
             }
 
+            var byteCount = encoding.GetByteCount(text);
+            var buffer = ArrayPool<byte>.Shared.Rent(byteCount);
+            try
+            {
+                encoding.GetBytes(text, 0, text.Length, buffer, 0);
+            }
+            catch (Exception)
+            {
+                // Make sure that the buffer will definitely be recycled
+                ArrayPool<byte>.Shared.Return(buffer);
+                throw;
+            }
```

Comment:
>  instead of returning separately in success and exceptional path

Would you please tell me the deep reason? I've read your example and I find if so, the `try` block will be much larger. I need to consider how to reconstruct them.

Reply:
It's what `finally` and `using` designed for.

> the `try` block will be much larger.

It's correct. `using` also introduces such large `try` scope.

Reply:
Done. Thanks for your explanation 🤝